catalog/index_spec.go

```diff
@@ -0,0 +1,20 @@
+package catalog
+
+import "time"
+
+type Index struct {
```

Comment:
I don't see those types being part of a public API so why are they in spec.go? Why are they public?

Reply:
Yes no need for them to be public. They could be in `types.go` or I'd be fine with having them in `catalog.go`.

Reply:
Sure, I will move it to catalog.go.